### ISSUE TITLE
Create table and repo for model revisions

### ISSUE THREAD

**Issue by leoll2** [open-edge-platform/training_extensions#4672]
Create the `model_revisions` table and relative repo, as per design.

<img width="287" height="279" alt="Image" src="https://github.com/user-attachments/assets/13613399-22bd-400e-b1ac-f3f6c548d877" />

Fields:
- `id`: unique identifier
- `project_id`: foreign key linking to the project that owns the model
- `architecture`: identifier of the model architecture (same as Geti manifest id)
- `parent_revision`: reference to the parent model revision (if any) that was fine-tuned to obtain this revision
- `training_status`: could be "not_started", "in_progress", "failed", "successful"
- `training_configuration`: object containing the full training configuration, hyperparameters, ...
- `training_dataset_id`: reference to the dataset revision used for training
- `label_schema_revision`: label schema used to train the model, encoded as JSON
- `training_started_at`: timestamp when the training job started
- `training_finished_at`: timestamp when the training job finished
- `files_deleted`: whether the binary files of this model are still available or not
